gapic/src/main/com/google/gapid/models/Devices.java

```diff
@@ -142,6 +153,14 @@ public void selectReplayDevice(Device.Instance dev) {
     listeners.fire().onReplayDeviceChanged(dev);
   }
 
+  public static void validateDevice(DeviceCaptureInfo device, Function<Boolean, Void> callback) {
+    DeviceValidationInfo.validateDevice(device, callback);
+  }
+
+  public static boolean getValidationStatus(DeviceCaptureInfo device) {
+    return DeviceValidationInfo.getValidationStatus(device);
+  }
+
```

Comment:
please no static singletons, please follow the pattern from my earlier review: have the model return a `ListenableFuture` of the result. Have the view code do the `UICallback` stuff. There are two patterns we follow in the models: event based (primarily for loading things) and by returning a `ListenableFuture`.

Reply:
I was following the event based pattern in this. Changed it to now to have the views do all the UICallback stuff and removed the static singletons as well.